lib/DDG/Spice/Symbolab.pm

```diff
@@ -2,17 +2,132 @@ package DDG::Spice::Symbolab;
 
 use DDG::Spice;
 
+my %legalWordsFunctions = (
+	"cos" => 2,
+	"sin" => 2,
+	"tan" => 2,
+	"cot" => 2,
+	"sec" => 2,
+	"csc" => 2,
+	
+	"arccos" => 2,
+	"arcsin" => 2,
+	"arctan" => 2,
+	"arccot" => 2,
+	"arcsec" => 2,
+	"arccsc" => 2,
+	
+	"cosh" => 2,
+	"sinh" => 2,
+	"tanh" => 2,
+	"coth" => 2,
+	"sech" => 2,
+	"csch" => 2,
+	
+	"sqrt" => 2,
+	"log" => 2,
+	"log10" => 2,
+	"ln" => 2,
+	
+	"calculate" => 1,
+	"solve" => 1,
+	"compute" => 1,
+	"integral" => 1,
+	"integrate" => 1,
+	"integration" => 1,
+	"antiderivative" => 1,
+	"derive" => 1,
+	"differentiate" => 1,
+	"derivation" => 1,
+	"derivative" => 1,
+	"factor" => 1,
+	"factorize" => 1,
+	"factorise" => 1,
+	"expand" => 1,
+    "x" => 1,                # to catch cases like x^2+2x=0
+    "2x" => 1, 
+    "3x" => 1,
+    "4x" => 1,
+    "5x" => 1,
+    "6x" => 1,
+    "7x" => 1,
+    "8x" => 1,
+    "9x" => 1
+);
+
 spice is_cached => 1;
 
 spice to => 'https://www.symbolab.com/ddg?query=$1';
 spice wrap_jsonp_callback => 1;
 
-triggers startend => "calculate", "solve", "compute", "integral", "integrate", "integration", "antiderivative", "derivative", "derive", "differentiate", "derivation";
+triggers startend => keys %legalWordsFunctions;
+
 handle query_lc => sub {
-    if ($_ =~ /[\^\*\=\+\-\/\\]/ or $_ =~ /\b(sin|cos|tan|cot|csc|sec|ln|log|sqrt)/i) {
-        return $_;
-    }
-    return;
+    #for cases like c++, j++
+	if ($_ =~ /\+\+/){  
+	  return;
+	}
+	
+	my $minWeight = 2;
```

Comment:
This can be defined outside the `handle` so we don't recreate the variable each time the `handle` function runs 👍